pyo3-build-config/src/impl_.rs

```diff
@@ -739,7 +810,7 @@ fn parse_script_output(output: &str) -> HashMap<String, String> {
 /// The sysconfigdata is simply a dictionary containing all the build time variables used for the
 /// python executable and library. Here it is read and added to a script to extract only what is
 /// necessary. This necessitates a python interpreter for the host machine to work.
-fn parse_sysconfigdata(sysconfigdata_path: impl AsRef<Path>) -> Result<InterpreterConfig> {
+pub fn parse_sysconfigdata(sysconfigdata_path: impl AsRef<Path>) -> Result<InterpreterConfig> {
```

Comment:
Hmm, I hate to be "that guy" who suggests a whole load of changes after you've done all this work. This just struck me and it seems really nice.

What if the signature of `parse_sysconfigdata` is 

```
pub fn parse_sysconfigdata(sysconfigdata_path: impl AsRef<Path>) -> Result<Sysconfigdata>

// where Sysconfigdata is just a wrapper around a hashmap:
struct Sysconfigdata(Hashmap<String, String>);
```

We could then have `Sysconfigdata::get_value(key: &str) -> Option<&str>` and `Interpreterconfig::from_sysconfigdata(sysconfigdata: Sysconfigdata) -> Result<InterpreterConfig>`.

Couple of reasons I suggest this:
 - PyO3 doesn't need to care about `abi_flags`, `abi_tag` and `ext_suffix` in the `InterpreterConfig` struct or its config files. Especially it seems slightly weird to have these in the config files even though they don't affect PyO3's build at all.
 - `maturin` can get those vars (and any others it wants in the future) directly from `Sysconfigdata.get_value`. Especially good is that then `maturin` doesn't need PyO3 to change if it wants extra sysconfigdata values in the future.
 - `IntepreterConfig::from_sysconfigdata` has a nice match with `InterpreterConfig::from_interpreter`

I'm sorry to suggest this so late 😢

Reply:
Hey no worries - you obviously know much more about the project so I appreciate the input and guidance. I'm not just trying to rack up merged PRs so I'm happy to take the time to get the interface right. Doubly so on a public API subject to SemVer, etc. I am learning along the way and that (along with helping) is my goal. Hopefully I am not holding things up by taking more time.

Anyway - this makes sense! I think the hashmap is basically what maturin uses now but without the `Sysconfigdata` wrapper. The simple struct basically says (to me) "this is a dict where you can make a pretty good guess about the keys, but we make no specific guarantees" which seems like a good fit for this file which can be quite long and varied.

Reply:
Great! Your contributions are warmly welcomed, you're not holding things up at all. Contributors are the lifeblood of OSS and if PyO3 will carry on should I ever be unable to keep working on it then a healthy group of knowledgeable contributors is even more important. (My intention is to continue with PyO3 until Rust and Python are dead languages, though I'm going to be a dad soon and you never know what life throws at you.) There's also way more things we want to improve in PyO3 than I as a single person have time for. People take time to familiarize with the project; no question is too silly!

I simply respect your time too and if I can suggest design ideas which influence a PR before too much work is spent on it, then I much prefer to do that! 🙂

Reply:
Well thanks - I think we're on the same page! I'll work on these changes next.